File: 409-longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        //Base Case
        if (s.length() <= 1){
            return s.length();
        }
        unordered_map<char,int> myMap;
        int odd = 0;
        for(int i=0; i<s.length(); i++){
            myMap[s[i]]++;
            if(myMap[s[i]] %2) odd++;
            else odd--;
        }
        if (odd) return s.length() - odd + 1;
        return s.length();
    }
};

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        char_count= Counter(s)
        print(char_count)
        output = 0
        odd = False
        for _, count in char_count.items():
            if count % 2 == 0:
                output += count
            else:
                if count - 1 != 0:
                    output += count-1
                    odd = True
                else:
                    odd = True
        if odd:
            output +=1            

        
        return output